cudaaligner/include/claragenomics/cudaaligner/aligner.hpp

```diff
@@ -58,7 +58,8 @@ class Aligner
     /// \param query_length  Query string length
     /// \param target Target string
     /// \param target_length Target string length
-    virtual StatusType add_alignment(const char* query, int32_t query_length, const char* target, int32_t target_length) = 0;
+    /// \param reverse_complement_target Reverse complement the target string
+    virtual StatusType add_alignment(const char* query, int32_t query_length, const char* target, int32_t target_length, bool reverse_complement_target = false) = 0;
```

Comment:
Can't the query also be a reverse complement?

Reply:
I think most use cases only reverse complement one of the strings, and generally the target. for the query reverse complement and target fixed, that's equivalent to query fixed + target reverse complemented + CIGAR reversed (but that's not very user friendly i guess).

to support both being reverse complemented, what do you think is better? (reverse_compelemnt_target, reverse_complement_query) or an enum that let's you choose (NONE, REVERSE_COMPLEMENT_TARGET, REVERSE_COMPLEMENT_QUERY, REVERSE_COMPLEMENT_BOTH)?

Reply:
Since query and target are independent entities and have such a property each, I would just do two arguments. Otherwise you will have to write a function which maps each combination to the enum when you loop over query and target pairs.

Reply:
Cool, I was thinking of the same.

Although for the enum it would be something like

```
if (REVERSE_COMPLEMENT_TARGET || BOTH)
{
    // reverse complement target when copying over
}
if (REVERSE_COMPLEMENT_QUERY || BOTH)
{
    // reverse complement target when copying over
}
```

Reply:
added

Reply:
I am thinking more about how to call the function than how implement it. So two bool arguments is preferable I'd say.